packages/ckeditor5-utils/src/dom/emittermixin.ts

```diff
@@ -45,116 +44,115 @@ import mix from '../mix';
-				capture: !!options.useCapture,
-				passive: !!options.usePassive
-			};
-
-			const proxyEmitter = this._getProxyEmitter( emitter, proxyOptions ) || new ProxyEmitter( emitter, proxyOptions );
-
-			this.listenTo( proxyEmitter, event, callback, options );
-		} else {
-			// Execute parent class method with Emitter (or ProxyEmitter) instance.
-			EmitterMixin.listenTo.call( this, emitter, event, callback, options );
+export default function DomEmitterMixin<Base extends abstract new( ...args: any[] ) => BaseEmitter>(
+	base: Base
+): {
+	new( ...args: ConstructorParameters<Base> ): InstanceType<Base> & Emitter;
+	prototype: InstanceType<Base> & Emitter;
+} {
+	abstract class Mixin extends base implements Emitter {
+		public override listenTo<K extends keyof HTMLElementEventMap>(
+			emitter: Node | Window,
+			event: K,
+			callback: ( this: this, ev: EventInfo, event: HTMLElementEventMap[ K ] ) => void,
+			options?: CallbackOptions & { readonly useCapture?: boolean; readonly usePassive?: boolean }
+		): void;
+		public override listenTo<TEvent extends BaseEvent>(
+			emitter: BaseEmitter,
+			event: TEvent[ 'name' ],
+			callback: GetCallback<TEvent>,
+			options?: CallbackOptions
+		): void;
+		public override listenTo(
+			emitter: BaseEmitter | Node | Window,
+			event: string,
+			callback: ( ev: EventInfo, ...args: any[] ) => void,
+			options: CallbackOptions & { readonly useCapture?: boolean; readonly usePassive?: boolean } = {}
+		): void {
+			// Check if emitter is an instance of DOM Node. If so, use corresponding ProxyEmitter (or create one if not existing).
+			if ( isNode( emitter ) || isWindow( emitter ) ) {
+				const proxyOptions = {
+					capture: !!options.useCapture,
+					passive: !!options.usePassive
+				};
+
+				const proxyEmitter = this._getProxyEmitter( emitter, proxyOptions ) || new ProxyEmitter( emitter, proxyOptions );
+
+				this.listenTo( proxyEmitter, event, callback, options );
+			} else {
+				// Execute parent class method with Emitter (or ProxyEmitter) instance.
+				BaseEmitter.prototype.listenTo.call( this, emitter, event, callback, options );
+			}
 		}
-	},
-
-	/**
-	 * Stops listening for events. It can be used at different levels:
-	 * It is backwards compatible with {@link module:utils/emittermixin~EmitterMixin#listenTo}.
-	 *
-	 * * To stop listening to a specific callback.
-	 * * To stop listening to a specific event.
-	 * * To stop listening to all events fired by a specific object.
-	 * * To stop listening to all events fired by all object.
-	 *
-	 * @param {module:utils/emittermixin~Emitter|Node|Window} [emitter] The object to stop listening to.
-	 * If omitted, stops it for all objects.
-	 * @param {String} [event] (Requires the `emitter`) The name of the event to stop listening to. If omitted, stops it
-	 * for all events from `emitter`.
-	 * @param {Function} [callback] (Requires the `event`) The function to be removed from the call list for the given
-	 * `event`.
-	 */
-	stopListening(
```

Comment:
This should probably not get lost and be moved to the interface method same as the above method.